File: content/1-basics/5-while-loop/rb/while_loop.rb
def main():
    print("Enter numbers (enter 0 to stop):")

    while True:
        number = int(input("Enter a number: "))

        if number == 0:
            break

        print("You entered:", number)

    print("Loop terminated because you entered 0.")

if __name__ == "__main__":
    main()

File: content/1-basics/5-while-loop/rs/while_loop.rs
use std::io;

fn main() {
    println!("Enter numbers (enter 0 to stop):");

    loop {
        let mut input = String::new();
        println!("Enter a number: ");
        io::stdin().read_line(&mut input).expect("Failed to read line");

        let number: i32 = input.trim().parse().expect("Please enter a valid number");

        if number == 0 {
            break;
        }

        println!("You entered: {}", number);
    }

    println!("Loop terminated because you entered 0.");
}